go-controller/pkg/ovn/namespace.go

```diff
@@ -351,14 +351,26 @@ func (oc *Controller) updateNamespace(old, newer *kapi.Namespace) {
 	aclAnnotation := newer.Annotations[aclLoggingAnnotation]
 	oldACLAnnotation := old.Annotations[aclLoggingAnnotation]
 	// support for ACL logging update, if new annotation is empty, make sure we propagate new setting
-	if aclAnnotation != oldACLAnnotation && (oc.aclLoggingCanEnable(aclAnnotation, nsInfo) || aclAnnotation == "") &&
-		len(nsInfo.networkPolicies) > 0 {
-		// deny rules are all one per namespace
-		if err := oc.setACLLoggingForNamespace(old.Name, nsInfo); err != nil {
-			klog.Warningf(err.Error())
-		} else {
-			klog.Infof("Namespace %s: ACL logging setting updated to deny=%s allow=%s",
-				old.Name, nsInfo.aclLogging.Deny, nsInfo.aclLogging.Allow)
+	if aclAnnotation != oldACLAnnotation && (oc.aclLoggingCanEnable(aclAnnotation, nsInfo) || aclAnnotation == "") {
+		if len(nsInfo.networkPolicies) > 0 {
+			// deny rules are all one per namespace
+			if err := oc.setACLLoggingForNamespace(old.Name, nsInfo); err != nil {
```

Comment:
is there a way to roll the egressFirewall updates into this function being that it is also part of the ACLLogging for the namespace? There should be a way to match all the ACLs we need without getting the EgressFirewall object. Could the update function search the cache for all ACLs with `ExternalIDs == map[string]string{"egressFirewall", namespace.Name}`

Reply:
I will post a fixup that I'll squash down into the main commit once you're o.k. with it, just to make reading easier.
I also had to rebase everything yet again on top of another commit that made it into master.

Indeed, the lookup for the egressfirewall object was not needed, but I'm against merging the egressfirewall policy logging update into `setACLLoggingForNamespace`. Instead, I renamed `setACLLoggingForNamespace` to: `setNetworkPolicyACLLoggingForNamespace` to clarify what that method does. If you look at the content of the method, you can see that it was meant exclusively for network policies and rebuilds the deny list in ~/development/ovn-kubernetes/go-controller/pkg/ovn/policy.go:269 in `buildDenyACLs` which is exclusive to network policy rules, and then it updates the network policy ACLs with:
`updateACLLoggingForPolicy`

But, the egressfw ACLs and network policy ACLs are 2 different, independent "things" and they even live at different locations in OVN.

The egress policy ACLs are attached to port groups:
~~~
sh-5.1# ovn-nbctl list port_group 
_uuid               : 3cd88080-609f-4db1-b025-ff9872d9c830
acls                : [b42036da-7016-4c5d-8b4c-458d27858fe3, e4dfeb2e-1d19-4f6e-b9a0-46d753304f44]
external_ids        : {name=a18007334074686647077_ingressDefaultDeny}
name                : a18007334074686647077_ingressDefaultDeny
ports               : [04af0632-f784-412e-a355-faa6bb515622, 6eaaf167-bef8-4aa7-901d-6ffa2ff5c0cb]

_uuid               : 313f4384-e6bb-4784-a7de-d9d5a5922712
acls                : []
external_ids        : {name=clusterRtrPortGroup}
name                : clusterRtrPortGroup
ports               : [05fae795-b7b0-40f1-857c-53740d8a9994, 4b6028ec-9e44-4c5a-9df3-ca844b51bd30, 867b43f8-849d-47b6-9084-c81b00cda539]

_uuid               : 922db569-89ab-446c-a62b-82739a5182e7
acls                : [30558740-d696-4519-a8c6-66c24ff4f443]
external_ids        : {name=test_allow-same-namespace}
name                : a17042190824695124102
ports               : [04af0632-f784-412e-a355-faa6bb515622, 6eaaf167-bef8-4aa7-901d-6ffa2ff5c0cb]

_uuid               : 4a60c2e1-da63-457e-a4fb-4e7b488931e6
acls                : [994c4ece-8f39-4a3d-9b39-c456bdd44497, b218e727-b555-43f3-a56c-beac7ec9e762]
external_ids        : {name=a18007334074686647077_egressDefaultDeny}
name                : a18007334074686647077_egressDefaultDeny
ports               : []

_uuid               : 1f2fc109-aead-4205-8040-87c7fe498416
acls                : []
external_ids        : {name=clusterPortGroup}
name                : clusterPortGroup
ports               : [08f04168-91e6-4009-8fdf-694025df0c8a, 1c96bd5a-138e-440f-ae97-4c38c82a0023, b35504ea-7ad7-4acb-8de3-69050c93216a]
sh-5.1# 
sh-5.1# 
sh-5.1# ovn-nbctl acl-list 922db569-89ab-446c-a62b-82739a5182e7
  to-lport  1001 (ip4.src == {$a3938097884066350661} && outport == @a17042190824695124102) allow-related log(name=test_allow-same-namespace_0,severity=notice,meter="acl-logging")
~~~

The egress firewall rules are attached to the join switch:
~~~
sh-5.1# ovn-nbctl acl-list ovn-worker2
  to-lport  1001 (ip4.src==10.244.1.2) allow-related
sh-5.1# ovn-nbctl acl-list join       
  to-lport 10000 ((ip4.dst == 172.18.0.0/16) && ip4.src == $a5154718082306775057 && inport == "jtor-ovn_cluster_router") allow
  to-lport  9999 ((ip4.dst == 10.96.0.0/16) && ip4.src == $a5154718082306775057 && inport == "jtor-ovn_cluster_router") allow
  to-lport  9998 ((ip4.dst == 10.244.0.0/16) && ip4.src == $a5154718082306775057 && inport == "jtor-ovn_cluster_router") allow
  to-lport  9997 ((ip4.dst == 0.0.0.0/0) && ip4.src == $a5154718082306775057 && inport == "jtor-ovn_cluster_router") drop
~~~

So I'd prefer to keep those 2 independent and to call them separately.